src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterOfT.cs

```diff
@@ -299,48 +313,30 @@ internal bool TryWriteDataExtensionProperty(Utf8JsonWriter writer, T value, Json
                 ThrowHelper.ThrowJsonException_SerializerCycleDetected(options.EffectiveMaxDepth);
             }
 
-            bool success;
             JsonDictionaryConverter<T> dictionaryConverter = (JsonDictionaryConverter<T>)this;
 
-            if (ClassType == ClassType.Value)
```

Comment:
So the original code here was not called for ClassType==Value? Or was this optimization not necessary?

Reply:
This branch was unreachable, as the converter is of type `JsonDictionaryConverter<T> ` which cannot be `ClassType.Value`. We branched into a separate path for custom (`ClassType.Value`) converters. This change should have been made in https://github.com/dotnet/runtime/pull/34569.